src/components/Api.js

```diff
@@ -0,0 +1,166 @@
+
+class Api {
+
+  constructor({ baseUrl, headers }) {
+    this.baseUrl = baseUrl;
+    this.headers = headers;
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._checkResponse);
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return response;
+  }
+
+  deleteCard(cardId) {
+    const url = `${this.baseUrl}/cards/${cardId}`;
+    const options = {
+      method: "DELETE",
+      headers: this.headers,
+    };
+
+    return this._request(url, options);
+  }
+
+  getInitialcards() {
+    const url = `${this.baseUrl}/cards`;
+    const options = {
+      method: "GET",
+      headers: this.headers,
+    };
+
+    return this._request(url, options).then((response) => {
+      return response.json();
```

Comment:
`response.json()` should be inside `_checkResponse` not to duplicate this parsing after every request